src/components/Microsoft.DotNet.UpgradeAssistant.Extensions/ExtensionAssemblyLoadContext.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions
+{
+    internal class ExtensionAssemblyLoadContext : AssemblyLoadContext
+    {
+        private readonly ExtensionInstance _extension;
+
+        public ExtensionAssemblyLoadContext(ExtensionInstance extension)
+            : base(extension.Name)
+        {
+            _extension = extension;
+        }
+
+        protected override Assembly? Load(AssemblyName assemblyName)
+        {
+            // If available in the default, we want to ensure that is used.
+            var inDefault = Default.Assemblies.Any(a => string.Equals(a.GetName().Name, assemblyName.Name, StringComparison.Ordinal));
+
+            if (inDefault)
+            {
+                return null;
+            }
+
+            var dll = $"{assemblyName.Name}.dll";
+            var dllFile = _extension.FileProvider.GetFileInfo(dll);
+
+            var pdb = $"{assemblyName.Name}.pdb";
```

Comment:
Super-nit-pick: You don't need to get the file info for the PDB if the dll doesn't exist, so these lines can move inside the first `if`.